From 6aa0c2864bbadf95412f1a6c49329b43bbcb0575 Mon Sep 17 00:00:00 2001
From: Tom Spurling <tom@tspurling.co.uk>
Date: Sat, 2 Jun 2018 11:58:03 +0100
Subject: [PATCH] Stop pulseaudio meddling

- PCM stays at -0dB
- Master stays where I've set it if I've turned it down, or -0dB otherwise, and doesn't jump randomly when headphones are (un)plugged
- Pulse volume maps to Headphone or Speaker volume, which is consistent with what it's trying to do. Master stays where it is.
- Master mute is left alone so that I can use it to mute while unplugging, without Pulse interfering and unmuting.
---
 analog-output-headphones.conf | 24 ++++++++--------
 analog-output-speaker.conf    | 53 +++++++++++++++++------------------
 analog-output.conf.common     |  2 +-
 3 files changed, 39 insertions(+), 40 deletions(-)

diff --git a/analog-output-headphones.conf b/analog-output-headphones.conf
index b6ee70b..717f632 100644
--- a/analog-output-headphones.conf
+++ b/analog-output-headphones.conf
@@ -53,19 +53,19 @@ state.unplugged = unknown
 required-any = any
 
 [Element Hardware Master]
-switch = mute
-volume = merge
+switch = ignore
+volume = ignore
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Master]
-switch = mute
-volume = merge
+switch = ignore
+volume = ignore
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Master Mono]
-switch = off
+switch = ignore
 volume = off
 
 [Element Speaker+LO]
@@ -74,21 +74,21 @@ volume = off
 
 [Element Headphone+LO]
 required-any = any
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Headphone]
 required-any = any
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Headset]
 required-any = any
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all
 override-map.2 = all-left,all-right
@@ -101,12 +101,12 @@ required-any = any
 ; the second headphones. But it should not hurt if we leave the second
 ; headphone jack enabled nonetheless.
 [Element Headphone2]
-switch = mute
+switch = ignore
 volume = zero
 
 [Element Speaker]
-switch = off
-volume = off
+switch = ignore
+volume = ignore
 
 [Element Desktop Speaker]
 switch = off
@@ -114,7 +114,7 @@ volume = off
 
 ; On some machines Front is actually a part of the Headphone path
 [Element Front]
-switch = mute
+switch = ignore
 volume = zero
 
 [Element Rear]
diff --git a/analog-output-speaker.conf b/analog-output-speaker.conf
index 9f4dac4..b1a2e1e 100644
--- a/analog-output-speaker.conf
+++ b/analog-output-speaker.conf
@@ -67,14 +67,14 @@ state.plugged = unknown
 state.unplugged = unknown
 
 [Element Hardware Master]
-switch = mute
-volume = merge
+switch = ignore
+volume = ignore
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Master]
-switch = mute
-volume = merge
+switch = ignore
+volume = ignore
 override-map.1 = all
 override-map.2 = all-left,all-right
 
@@ -82,11 +82,10 @@ override-map.2 = all-left,all-right
 switch = off
 volume = off
 
-; This profile path is intended to control the speaker, let's mute headphones
-; else there will be a spike when plugging in headphones
+; leave it alone please
 [Element Headphone]
-switch = off
-volume = off
+switch = ignore
+volume = ignore
 
 [Element Headphone2]
 switch = off
@@ -98,124 +97,124 @@ volume = off
 
 [Element Speaker+LO]
 required-any = any
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Speaker]
 required-any = any
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Desktop Speaker]
 required-any = any
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all
 override-map.2 = all-left,all-right
 
 [Element Front]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-front
 override-map.2 = front-left,front-right
 
 [Element Front Speaker]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-front
 override-map.2 = front-left,front-right
 required-any = any
 
 [Element Speaker Front]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-front
 override-map.2 = front-left,front-right
 required-any = any
 
 [Element Rear]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-rear
 override-map.2 = rear-left,rear-right
 
 [Element Surround]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-rear
 override-map.2 = rear-left,rear-right
 
 [Element Surround Speaker]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-rear
 override-map.2 = rear-left,rear-right
 required-any = any
 
 [Element Speaker Surround]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-rear
 override-map.2 = rear-left,rear-right
 required-any = any
 
 [Element Side]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-side
 override-map.2 = side-left,side-right
 
 [Element Speaker Side]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-side
 override-map.2 = side-left,side-right
 
 [Element Center]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-center
 override-map.2 = all-center,all-center
 
 [Element Center Speaker]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-center
 override-map.2 = all-center,all-center
 required-any = any
 
 [Element LFE]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = lfe
 override-map.2 = lfe,lfe
 
 [Element LFE Speaker]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = lfe
 override-map.2 = lfe,lfe
 required-any = any
 
 [Element Bass Speaker]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = lfe
 override-map.2 = lfe,lfe
 required-any = any
 
 [Element CLFE]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-center
 override-map.2 = all-center,lfe
 
 [Element Speaker CLFE]
-switch = mute
+switch = ignore
 volume = merge
 override-map.1 = all-center
 override-map.2 = all-center,lfe
diff --git a/analog-output.conf.common b/analog-output.conf.common
index 17b4527..e15ec55 100644
--- a/analog-output.conf.common
+++ b/analog-output.conf.common
@@ -125,7 +125,7 @@
 
 [Element PCM]
 switch = mute
-volume = merge
+volume = zero
 override-map.1 = all
 override-map.2 = all-left,all-right
 
-- 
2.17.1

